optimum/intel/openvino/quantization.py

```diff
@@ -165,14 +165,29 @@ def __init__(
         self.apply_caching = apply_caching
         self.inference_result_mock = inference_result_mock
         self.tensor_cache = {}
+        self.stateful = len(request.query_state()) > 0
+        self._reset_state_called = False
 
     def collect_inputs(self, inputs):
+        if self.stateful:
+            if isinstance(inputs, dict) and is_nncf_version(">", "2.19"):
+                from nncf.definitions import NNCF_DATASET_RESET_STATE_KEY
+
+                # To reflect the state resetting during NNCF calibration, we add a special key to the input dict
+                # Shallow copying is done on purpose shallow copy on purpose: we only need to add a key to the top-level dict
+                inputs = inputs.copy()
+                inputs[NNCF_DATASET_RESET_STATE_KEY] = self._reset_state_called
+            self._reset_state_called = False
```

Comment:
would you mind adding a comment to state when `inputs[NNCF_DATASET_RESET_STATE_KEY]` should be set to True or False, I'm not sure to understand as I see that `self._reset_state_called` is set to True when `reset_state` is called + `reset_state` being called when `inputs[NNCF_DATASET_RESET_STATE_KEY] is True` https://github.com/openvinotoolkit/nncf/blob/c7afa57fa6910883d5595db66c5785059eb9cb11/src/nncf/openvino/engine.py#L52

Reply:
> thanks for adding a comment!

As always no problem!

> my last question is why do we have 2 * n_samples and not n_samples ?

During `.genereate()` call of Whisper pipeline, the first decoder call is actually a language detection call. Only then decoder starts generating actual text tokens. So for each sample, state is reset the first time before the language detection forward, and the second time before token generation phase.